### PR TITLE
Adding an example for docker and docker-compose, because it's 2016 :D

### PR DESCRIPTION
**Pull request by unlucio**

### CODE REVIEW

**Comment by unlucio** [comment]
@davidmarkclements I was preparing a convenient docker image for use to use for profiling and thought to share it since it can be an example for 0x.
Idk if it can be useful

**Comment by mcollina** [comment]
Does this support OS X, or it is just Linux? In case, just put it in the README :)

**Comment by unlucio** [comment]
well, I'm running it on osx (in a virtual machine :D), tbt I'm not sure how it would react to boot3docker :P
but there's a [note](https://github.com/davidmarkclements/0x/pull/9/files#diff-0730bb7c2e8f9ea2438b52e419dd86c9R277)

**Comment by davidmarkclements** [comment]
Yah this is awesome, but for a merge it would be great if we could go beyond the note and verify support via docker-machine (aka boot2docker)

**Comment by unlucio** [comment]
can someone using boot2docker verify it for me? :P

**Comment by davidmarkclements** [comment]
so this doesn't work on OS X, because the kernel for boot2docker has no corresponding linux-tools package

```
a024ee45ad_docker_flames_1 | /usr/bin/perf: line 32: lsb_release: command not found
a024ee45ad_docker_flames_1 | WARNING: perf not found for kernel 4.1.13
a024ee45ad_docker_flames_1 |
a024ee45ad_docker_flames_1 |   You may need to install the following packages for this specific kernel:
a024ee45ad_docker_flames_1 |     linux-tools-4.1.13-boot2docker
a024ee45ad_docker_flames_1 |     linux-cloud-tools-4.1.13-boot2docker
a024ee45ad_docker_flames_1 |
a024ee45ad_docker_flames_1 |   You may also want to install one of the following packages to keep up to date:
a024ee45ad_docker_flames_1 |     linux-tools-boot2docker
a024ee45ad_docker_flames_1 |     linux-cloud-tools-boot2docker
a024ee45ad_docker_flames_1 exited with code 2
```

Those packages do not exist

@unlucio can we change the readme to say Linux Only

File: readme.md [hunk 1]
```diff
@@ -264,6 +264,27 @@ Now try some other options, e.g.
...
 ```
 
+### Or using [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/)
+_this process has been tested on linux, different milage may apply on OSX with boot2docker_
+Clone this repo and `cd` to the `examples/docker` folder
+
+```
+docker-compose up
+```
+open another shell and send you SIGINT:
```
Comment (davidmarkclements) [inline readme.md]: send a SIGINT

File: readme.md [hunk 2]
```diff
@@ -264,6 +264,27 @@ Now try some other options, e.g.
...
+
+```
+sudo kill -s SIGINT `ps aux | grep 'node /usr/local/bin/0x' | head -1| awk '{print $2}'`
+# might differ for boot2docker
```
Comment (davidmarkclements) [inline readme.md]: remove might differ

File: readme.md [hunk 3]
```diff
@@ -264,6 +264,27 @@ Now try some other options, e.g.
 0x -t examples/rest-api
 ```
 
+### Or using [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/)
+_this process has been tested on linux, different milage may apply on OSX with boot2docker_
```
Comment (davidmarkclements) [inline readme.md]: remove, doesn't work on OS X

File: readme.md [hunk 4]
```diff
@@ -264,6 +264,27 @@ Now try some other options, e.g.
...
+Clone this repo and `cd` to the `examples/docker` folder
+
+```
+docker-compose up
+```
+open another shell and send you SIGINT:
+
+```
+sudo kill -s SIGINT `ps aux | grep 'node /usr/local/bin/0x' | head -1| awk '{print $2}'`
```
Comment (davidmarkclements) [inline readme.md]: wait, processes in containers show up in ps aux on linux?

Comment (unlucio) [inline readme.md]: yup :)
and if the container lacks of a basic init system the zombies might even end up mingling around attached to the host's init process

File: readme.md (hunk 1 above)
Comment (unlucio) [inline readme.md]: docker-compose doesn't work on osx? O.o

File: readme.md (hunk 3 above)
Comment (unlucio) [inline readme.md]: docker-compose doesn't work on osx? O.o

**Comment by unlucio** [comment]
BTW I subscribed the bets for the new [docker for mac](https://blog.docker.com/2016/03/docker-for-mac-windows-beta/) so I'll give a spin to this asap I get news about it.

**Comment by davidmarkclements** [comment]
just for the record, in case any one else thinks to try this, I overwrote /bin/uname in the docker container with a script that output 4.2.0-16-generic so it lies to perf. This made perf run, however no stacks were collected :(

**Comment by davidmarkclements** [comment]
WOW docker for mac looks awesome

File: readme.md (hunk 4 above)
Comment (davidmarkclements) [inline readme.md]: yes I've heard about that

Comment (unlucio) [inline readme.md]: well, is not that "on osx doesn't happen". It's just that they stay inside the vm :)

**Comment by unlucio** [comment]
mumble: the docker image created by this PR should come form ` ubuntu:wily` and therefore install the correct perfs tool. That's one of the problems I had with pre packaged node images.

**Comment by davidmarkclements** [comment]
but the docker kernel on os x doesn't have a corresponding perf tool

**Pull request closed without merging**